run_dir/static/js/running_notes.js

```diff
@@ -8,25 +8,26 @@ $(function(){
 });
 
 function generate_category_label(category){
-    if (category == 'Workset'){
-         category='<span class="label label-primary">'+ category +"</span>";
-     }else if (category == 'Flowcell'){
-         category='<span class="label label-success">'+ category +"</span>";
-     }else if (category == 'Meeting' || category == "Decision"){
-         category='<span class="label label-info">'+ category +"</span>";
-     }else if (category == 'User Communication'){
-         category='<span class="label label-danger">'+ category +"</span>";
-     }else if (category == 'Bioinformatics'){
-         category='<span class="label label-warning">'+ category +"</span>";
-     }else if (category == 'Important'){
-         category='<span class="label label-imp">'+ category +"</span>";
-     }else if (category == 'Deviation'){
-	       category='<span class="label label-devi">'+ category +"</span>";
-     }else if (category != ''){
-         category='<span class="label label-default">'+ category +"</span>";
-     }
-     return category;
+   if (category == 'Decision'){
+        category='<span class="label label-info">'+ category +"</span>";
+    }else if (category == 'Lab'){
+        category='<span class="label label-success">'+ category +"</span>";
+    }else if (category == 'Bioinformatics'){
+        category='<span class="label label-warning">'+ category +"</span>";
+    }else if (category == 'User Communication'){
+        category='<span class="label label-usr">'+ category +"</span>";
+    }else if (category == 'Administration'){
+        category='<span class="label label-danger">'+ category +"</span>";
+    }else if (category == 'Important'){
+        category='<span class="label label-imp">'+ category +"</span>";
+    }else if (category == 'Deviation'){
+        category='<span class="label label-devi">'+ category +"</span>";
+    }else if (category != ''){
+        category='<span class="label label-default">'+ category +"</span>";
+    }
+    return category;
```

Comment:
Suggested refactoring - reduces the lines of code a bit, but mostly makes it a lot more readable than a huge list of `else if` statements 😄 

```suggestion
    var cat_classes = {
        'Decision': 'info',
        'Lab': 'success',
        'Bioinformatics': 'warning',
        'User Communication': 'usr',
        'Administration': 'danger',
        'Important': 'imp',
        'Deviation': 'devi'
    }
    // Default button formatting
    category='<span class="label label-default">'+ category +"</span>";
    // Check if we recognise this category in the class object keys
    if(Object.keys(cat_classes).indexOf(category) != -1){
        category='<span class="label label-'+ cat_classes[category] +'">'+ category +"</span>";
    }
    return category;
```

Reply:
Hmm I'm getting an `llegal return statement`

Reply:
Difficult to help without seeing the code in context, but that often happens if you're missing a closing bracket, or a quote mark etc somewhere.